web/Dockerfile

```diff
@@ -67,24 +98,51 @@ ENV PYTHONDONTWRITEBYTECODE 1
 ENV PYTHONUNBUFFERED 1
 
 # Download Go packages
-RUN go install -v github.com/jaeles-project/gospider@latest
-RUN go install -v github.com/tomnomnom/gf@latest
-RUN go install -v github.com/tomnomnom/unfurl@latest
-RUN go install -v github.com/tomnomnom/waybackurls@latest
-RUN go install -v github.com/projectdiscovery/httpx/cmd/httpx@latest
-RUN go install -v github.com/projectdiscovery/subfinder/v2/cmd/subfinder@latest
-RUN go install -v github.com/projectdiscovery/nuclei/v3/cmd/nuclei@latest
-RUN go install -v github.com/projectdiscovery/naabu/v2/cmd/naabu@latest
-RUN go install -v github.com/hakluke/hakrawler@latest
-RUN go install -v github.com/lc/gau/v2/cmd/gau@latest
-RUN go install -v github.com/jaeles-project/gospider@latest
-RUN go install -v github.com/owasp-amass/amass/v3/...@latest
-RUN go install -v github.com/ffuf/ffuf@latest
-RUN go install -v github.com/projectdiscovery/tlsx/cmd/tlsx@latest
-RUN go install -v github.com/hahwul/dalfox/v2@latest
-RUN go install -v github.com/projectdiscovery/katana/cmd/katana@latest
-RUN go install -v github.com/dwisiswant0/crlfuzz/cmd/crlfuzz@latest
-RUN go install -v github.com/sa7mon/s3scanner@latest
+
+
+
+RUN ARCH=$(dpkg --print-architecture) \
+    && if [ "$ARCH" = "arm64" ]; then \
```

Comment:
Instead of listing out all the Go modules twice, just pass your `$ARCH` variable to `GOARCH`:

```bash
RUN ARCH=$(dpkg --print-architecture) \
    && if [[ "$ARCH" = "arm64" || "$ARCH" = "amd64" ]]; then
        GOARCH=$ARCH go install -v github.com/jaeles-project/gospider@latest \
        && GOARCH=$ARCH go install -v github.com/tomnomnom/gf@latest \
        && GOARCH=$ARCH go install -v github.com/tomnomnom/unfurl@latest \
        && GOARCH=$ARCH go install -v github.com/tomnomnom/waybackurls@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/httpx/cmd/httpx@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/subfinder/v2/cmd/subfinder@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/nuclei/v3/cmd/nuclei@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/naabu/v2/cmd/naabu@latest \
        && GOARCH=$ARCH go install -v github.com/lc/gau/v2/cmd/gau@latest \
        && GOARCH=$ARCH go install -v github.com/jaeles-project/gospider@latest \
        && GOARCH=$ARCH go install -v github.com/owasp-amass/amass/v3/...@latest \
        && GOARCH=$ARCH go install -v github.com/ffuf/ffuf@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/tlsx/cmd/tlsx@latest \
        && GOARCH=$ARCH go install -v github.com/hahwul/dalfox/v2@latest \
        && GOARCH=$ARCH go install -v github.com/projectdiscovery/katana/cmd/katana@latest \
        && GOARCH=$ARCH go install -v github.com/dwisiswant0/crlfuzz/cmd/crlfuzz@latest \
        && GOARCH=$ARCH go install -v github.com/sa7mon/s3scanner@latest; \
    else \
        echo "Unknown architecture: $ARCH" ; \
        exit 1; \
    fi
```

Reply:
Thank you for your feedback. I have optimized it